addons/snailmail/models/snailmail_letter.py

```diff
@@ -363,9 +363,8 @@ def fetch_failed_letters(self):
 
     @api.model
     def _is_valid_address(self, record):
-        record.ensure_one()
         required_keys = ['street', 'city', 'zip', 'country_id']
-        return all(record[key] for key in required_keys)
+        return len(record) <= 1 and all(record[key] for key in required_keys)
```

Comment:
Hello, this will return a potentially incorrect value when it is called on multiple records instead of an error.

Maybe you just want a preventing return before `ensure_one` if it's specifically equal to 0, but even then this does not look like a good idea.

I think the correct fix should be to not call this method in the first place when in the problematic case.